src/extension/wps-download/src/test/java/org/geoserver/wps/gs/download/DownloadMapProcessTest.java

```diff
@@ -248,9 +249,49 @@ public void downloadMapMars() throws Exception {
 
     @Test
     public void downloadRemoteSimple11() throws Exception {
-        String request = getTestRequest("mapRemoteSimple11.xml");
-        String caps111 = getTestRequest("caps111.xml");
-        byte[] getMapBytes = FileUtils.readFileToByteArray(new File(SAMPLES + "mapSimple.png"));
+        byte[] bytes =
+                testRemoteSimple(
+                        "mapSimple.png", "mapRemoteSimple11.xml", "caps111.xml", "image/png");
+        BufferedImage image = ImageIO.read(new ByteArrayInputStream(bytes));
+        ImageAssert.assertEquals(new File(SAMPLES + "mapSimple.png"), image, 100);
+    }
+
+    @Test
+    public void downloadRemoteSimple11Opacity() throws Exception {
+        byte[] bytes =
+                testRemoteSimple(
+                        "mapSimpleWithAlpha.png",
+                        "mapRemoteSimple11Opacity.xml",
+                        "caps111.xml",
+                        "image/png");
+        BufferedImage image = ImageIO.read(new ByteArrayInputStream(bytes));
+        ImageAssert.assertEquals(new File(SAMPLES + "mapSimpleWithAlpha.png"), image, 100);
```

Comment:
This bit works, and I've verified that the mapSimpleWithAlpha has 128 in the alpha channel, so the output should have 50% of it. What's confusing is the ImageAssert passing anyways, even if it's comparing against the original image. I guess it happens because the ImageAssert was built to be tolerant of antialising differences between images, which does use various degrees of alpha channel value to achieve its goal (and thus the image assert tolerates differences in alpha values, I presume).

That said... I had to try it out by hand to figure out what was going on, some explanation here in the test would be good.

Reply:
comments added